### ISSUE TITLE
ui doesn't load when searching for a trace id with leading zero

### ISSUE THREAD

**Issue by kung-foo** [jaegertracing/jaeger-ui#236]
## Requirement - what kind of business use case are you trying to solve?

Search for a trace with a leading zero (i.e. `07096c3e501cfcd70ffeb10e0c5874fe`)

## Problem - what in Jaeger blocks you from solving the requirement?

UI goes into infinite loop requesting `/api/traces/07096c3e501cfcd70ffeb10e0c5874fe`

Note: the api request _does_ succeed (as does a request without the leading zero)

Build: container jaegertracing/all-in-one
Trace sources: go.opencensus.io v0.14.0

**Comment by yurishkuro** [comment]
@tiffon I thought we already fixed it once.

**Comment by kung-foo** [comment]
```docker run --rm -it jaegertracing/all-in-one version
{"gitCommit":"75af597442cd2fe72eb88975e488354d97b62dd9","GitVersion":"v1.6.0","BuildDate":"2018-08-22T10:19:37Z"}
```

**Comment by everett980** [comment]
I believe this can be closed as updating to `1.7.0` or later seems to resolve this issue (tested on minor versions `1.7.0` - `1.12.0` (current latest)).

I was able to reproduce by firing up `jagertracing/all-in-one:1.6` and [Hot R.O.D. from source](https://github.com/jaegertracing/jaeger/tree/master/examples/hotrod).
In the Hot R.O.D. tab I generated 150 traces through:
```
{
    const btn = $('.btn')[0];
    let clicks = 0;
    const interval = setInterval(() => {
        btn.click();
        if (++clicks > 150) clearInterval(interval);
    }, 3);
}
```
I searched for [traces in the last hour with a limit of 1500](http://localhost:16686/search?end=1558378905629000&limit=1500&lookback=1h&maxDuration&minDuration=1ms&service=frontend).
Since leading zeros are stripped out of `traceID`s, I found `traceID`s that were shorter through:
```
$$('[href^="/trace/"]').reduce((map, elem) => ({
  ...map,
  [elem.href.length]: Reflect.has(map, elem.href.length) ? [...map[elem.href.length], elem.href] : [elem.href],
}));
```
On `1.6`, navigating to `/trace/${shortTraceID or fullLengthTraceID}` would work, but navigating to `/trace/0+${shortTraceID or fullLengthTraceID}` would hang. On `1.7` or newer, it works regardless of the number of leading zeros.

I issued a [PR to `jaegertracing/jaeger`](https://github.com/jaegertracing/jaeger/pull/1555) to update the Hot R.O.D. docs to use `latest` instead of `1.6`.

Please re-open this issue if it is reproducible on the more current version!
